### ISSUE TITLE
Headers are not able to add

### ISSUE THREAD

**Issue by RaruRv** [werk85/fetch-intercept#49]
This might be an issue with the lack of documentation. I am not able to add headers.

**Comment by iSanjayAchar** [comment]
Yes, there is a lack of documentation. Below solution is working for me

```
import fetchIntercept from 'fetch-intercept';
import { env } from './../const';
import AsyncStorage from '@react-native-community/async-storage';

export const unregister = fetchIntercept.register({
    request: async (url, config) => {
        const token = await AsyncStorage.getItem('auth_token');
        const headers = {
            'Content-Type': 'application/json',
            'Authorization': `Token ${ token }`,
            'Accept': 'application/json'
        }

        return [`${env.endpoint}${url}`, { headers, ...config }];
    },
    response: (response) => {
        return response;
    },
    responseError: (error) => {
        return Promise.reject(error);
    }
});
```

**Comment by RaruRv** [comment]
Thankyou for your response, I had solved the issue like :

`
 import fetchIntercept from 'fetch-intercept';
import { env } from './../const';
import AsyncStorage from '@react-native-community/async-storage';

export const unregister = fetchIntercept.register({
    request: async (url, config) => {
        const token = await AsyncStorage.getItem('auth_token');
        const headers = {
           config.headers = {Authorization: token}
        }

        return [`${env.endpoint}${url}`, { headers, ...config }];
    },
    response: (response) => {
        return response;
    },
    responseError: (error) => {
        return Promise.reject(error);
    }
});
`

**Comment by vedic-tech** [comment]
I solved the issue like below in my RN App,

`import fetchIntercept from 'fetch-intercept';

const fetchInterceptRegister = fetchIntercept.register({
  request: function(url, config) {
    const modifiedHeaders = new Headers(config.headers);
    modifiedHeaders.append('header_key_name', 'header_value');
    modifiedHeaders.append('header_key_name', "header_value");
    config.headers = modifiedHeaders;

    return [url, config];
  },

  requestError: function(error) {
    // Called when an error occured during another 'request' interceptor call
    return Promise.reject(error);
  },

  response: function(response) {
    // Modify the reponse object
    return response;
  },

  responseError: function(error) {
    // Handle an fetch error
    return Promise.reject(error);
  },
});

export default fetchInterceptRegister;`

**Comment by tabaktoni** [comment]
The issue is that it overrides existing headers, there is no way to happen to default fetch headers.
